lib/jsonapi/utils/support/pagination.rb

```diff
@@ -44,12 +44,7 @@ def pagination_params(records, options)
         #
         # @api private
         def paginator
-          @paginator ||=
-            if JSONAPI.configuration.default_paginator == :paged
-              PagedPaginator.new(page_params)
-            elsif JSONAPI.configuration.default_paginator == :offset
-              OffsetPaginator.new(page_params)
-            end
+          @paginator ||= "#{JSONAPI.configuration.default_paginator}_paginator".classify.constantize.new(page_params)
```

Comment:
What do you think on taking the class constant build part into another method?! It simplifies the readability (and probably debugging) quite a bit.

```ruby
def paginator
  @paginator ||= paginator_klass.new(page_params)
end

def paginator_klass
  "#{JSONAPI.configuration.default_paginator}_paginator".classify.constantize
end
```